### PR TITLE
console: Insert 'prioritySearch' function in jsUtils to enhance search functionality

### LINKED ISSUE #4321: Enhance Search functionality of tables in Data & Schema Management
The search functionality in the Data & Schema Management of the Hasura seems to filter records considering the _case of letters_ as well.
If this was an intended functionality, it's still better to show the tables which match the filter condition and don't match the letter-case below the tables that match with the exact letter-case.

### PR DESCRIPTION
**Pull request by Varun-Choudhary**
closes #4321 

### Description
This PR enhance the search functionality of Data, Actions, Remote Schemas and Events by introducing a `prioritySearch` function in `jsUtils`

### Affected components
- [x] Console

### CODE REVIEW

File: console/src/components/Common/utils/jsUtils.tsx [hunk 1]
```diff
@@ -353,3 +353,40 @@ export const getCurrTimeForFileName = () => {
...
+    });
+    searchResults = [
+      ...primarySameCaseResults,
+      ...secondarySameCaseResults,
+      ...primaryOppCaseResults,
+      ...secondaryOppCaseResults,
+    ];
+  } else {
+    searchResults = [...searchList];
+  }
```
Comment (kolharsam) [inline console/src/components/Common/utils/jsUtils.tsx:390]: In my opinion, I feel it is better to handle these cases, before coming to the main part of the function. So my suggestion would be to just initialize the variable `searchResults` with `searchList` itself and eliminate this `else` case

File: console/src/components/Common/utils/jsUtils.tsx [hunk 2]
```diff
@@ -353,3 +353,40 @@ export const getCurrTimeForFileName = () => {
...
+      const currentTerm = term[searchProp];
+      if (currentTerm.startsWith(searchText)) {
+        primarySameCaseResults.push(term);
+      } else if (currentTerm.includes(searchText)) {
+        secondarySameCaseResults.push(term);
+      } else if (
+        currentTerm.toLowerCase().startsWith(searchText.toLowerCase())
+      ) {
+        primaryOppCaseResults.push(term);
+      } else if (currentTerm.toLowerCase().includes(searchText.toLowerCase())) {
```
Comment (kolharsam) [inline console/src/components/Common/utils/jsUtils.tsx:378]: Here `currentTerm.toLowerCase()` and `searchText.toLowerCase()` is used repeatedly. So why don't we put them into separate variables?

File: console/src/components/Common/utils/jsUtils.tsx (hunk 1 above)
Comment (Varun-Choudhary) [reply console/src/components/Common/utils/jsUtils.tsx:390]: Yeah. Got it! Will make the changes.

File: console/src/components/Common/utils/jsUtils.tsx (hunk 2 above)
Comment (Varun-Choudhary) [reply console/src/components/Common/utils/jsUtils.tsx:378]: Cool. Will do this.

**Comment by Stefmore02** [comment]
Closing this PR as it's not relevant anymore.

**Pull request closed without merging**